diff --git a/addressfield_phone.module b/addressfield_phone.module
index 66692af..f47b463 100644
--- a/addressfield_phone.module
+++ b/addressfield_phone.module
@@ -310,17 +310,35 @@ function addressfield_phone_form_field_ui_field_edit_form_alter(&$form, $form_st
     $field = field_info_instance($entity_type, $field_name, $bundle);
     $settings = $field['widget']['settings'];
     $form['instance']['widget']['settings']['phone_number_fields'] = array(
-      '#type' => 'checkboxes',
+      '#type' => 'fieldset',
       '#title' => t('Available Phone Number Fields'),
-      '#description' => t('If no phone fields are selected, all phones fields will be available.'),
-      '#options' => _addressfield_phone_number_options_list(),
+      '#description' => t('If no phone fields are selected, all phones fields will be available as optional.'),
       '#states' => array(
         'visible' => array(// action to take.
           ':input[name="instance[widget][settings][format_handlers][phone]"]' => array('checked' => TRUE),
         ),
       ),
-      '#default_value' => $settings['phone_number_fields'],
     );
+
+    $field_options = array(
+      '' => t('Hidden'),
+      'optional' => t('Optional'),
+      'required' => t('Required'),
+    );
+
+    foreach (_addressfield_phone_number_options_list() as $number => $label) {
+      $value = '';
+      if (!empty($settings['phone_number_fields'][$number])) {
+        $value = $settings['phone_number_fields'][$number];
+      }
+
+      $form['instance']['widget']['settings']['phone_number_fields'][$number] = array(
+        '#type' => 'select',
+        '#title' => $label,
+        '#options' => $field_options,
+        '#default_value' => $value,
+      );
+    }
   }
 }
 
diff --git a/plugins/format/phone.inc b/plugins/format/phone.inc
index dd4d1de..f15ec65 100644
--- a/plugins/format/phone.inc
+++ b/plugins/format/phone.inc
@@ -36,7 +36,7 @@ function addressfield_format_phone_generate(&$format, $address, $context = array
       $settings = array();
     }
 
-    if (!$settings || isset($settings['phone'])) {
+    if (!$settings || !empty($settings['phone'])) {
       $format['phone_block']['phone_number']['phone_number'] = array(
         '#title' => t('Contact Phone'),
         '#size' => 15,
@@ -44,9 +44,10 @@ function addressfield_format_phone_generate(&$format, $address, $context = array
         '#type' => 'textfield',
         '#tag' => 'span',
         '#default_value' => isset($address['phone_number']) ? $address['phone_number'] : '',
+        '#required' => (!empty($settings['phone']) && $settings['phone'] == 'required'),
       );
     }
-    if (!$settings || isset($settings['extension'])) {
+    if (!$settings || !empty($settings['extension'])) {
       $format['phone_block']['phone_number']['phone_number_extension'] = array(
         '#title' => t('Phone Extension'),
         '#size' => 10,
@@ -54,9 +55,10 @@ function addressfield_format_phone_generate(&$format, $address, $context = array
         '#type' => 'textfield',
         '#tag' => 'span',
         '#default_value' => isset($address['phone_number_extension']) ? $address['phone_number_extension'] : '',
+        '#required' => (!empty($settings['extension']) && $settings['extension'] == 'required'),
       );
     }
-    if (!$settings || isset($settings['mobile'])) {
+    if (!$settings || !empty($settings['mobile'])) {
       $format['phone_block']['mobile_number'] = array(
         '#title' => t('Mobile Number'),
         '#size' => 15,
@@ -64,9 +66,10 @@ function addressfield_format_phone_generate(&$format, $address, $context = array
         '#type' => 'textfield',
         '#tag' => 'div',
         '#default_value' => isset($address['mobile_number']) ? $address['mobile_number'] : '',
+        '#required' => (!empty($settings['mobile']) && $settings['mobile'] == 'required'),
       );
     }
-    if (!$settings || isset($settings['fax'])) {
+    if (!$settings || !empty($settings['fax'])) {
       $format['phone_block']['fax_number'] = array(
         '#title' => t('Fax Number'),
         '#size' => 15,
@@ -74,6 +77,7 @@ function addressfield_format_phone_generate(&$format, $address, $context = array
         '#type' => 'textfield',
         '#tag' => 'div',
         '#default_value' => isset($address['fax_number']) ? $address['fax_number'] : '',
+        '#required' => (!empty($settings['fax']) && $settings['fax'] == 'required'),
       );
     }
   }
